app/src/main/java/com/cornellappdev/uplift/data/auth/ApolloAuthInterceptor.kt

```
@@ -0,0 +1,76 @@
+                    if (newAccessToken != null) {
+                        tokenManager.saveTokens(newAccessToken, refreshToken)
+                        // Retry the request with the new token
+                        val newRequest = request.newBuilder()
+                            .addExecutionContext(RetryContext(retryCount + 1))
+                            .build()
+                        emitAll(chain.proceed(newRequest))
+                        return@flow
+                    } else {
+                        sessionManager.logout()
+                    }
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)